### PR TITLE
Start recording tests and add intial Session recording

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

**Review by drwill-ms — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/IotHubServiceTestEnvironment.cs [hunk 1]
```diff
@@ -12,5 +12,7 @@ public IotHubServiceTestEnvironment()
             : base("iot")
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/IotHubServiceTestEnvironment.cs:12]: Can you update to using the variable set in TestSettings?

**Review by barustum — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/E2eTestBase.cs [hunk 2]
```diff
@@ -34,8 +36,18 @@ public virtual void SetupE2eTestBase()
 
         protected IoTHubServiceClient GetClient()
         {
+            string connectionString = TestEnvironment.IotHubConnectionString;
+
+            // In playback mode we will restore the shared access key to an invalid value so the connection string can be parsed.
+            if (Recording.Mode == RecordedTestMode.Playback)
+            {
+                connectionString = connectionString.Replace(";SharedAccessKey=", ";SharedAccessKey=Kg==;");
```
Comment (pakrym) [inline sdk/iot/Azure.Iot.Hub.Service/tests/E2eTestBase.cs:44]: Use the "Kg==;" value in SanitizeConnectionString

**Review by pakrym — APPROVED** [review]
LGTM

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestConnectionStringSanitizer.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.TestFramework;
+using Azure.Core;
+
+namespace Azure.Iot.Hub.Service.Tests
+{
+    internal class TestConnectionStringSanitizer : RecordedTestSanitizer
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestConnectionStringSanitizer.cs:9]: Looks good. Just check with Pavel and go ahead.

**Review by vinagesh — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/TestConnectionStringSanitizer.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                TestsConstants.IOT_HUB_CONNECTION_STRING => SanitizeConnectionString(environmentVariableValue),
+                _ => base.SanitizeVariable(variableName, environmentVariableValue)
+            };
+
+        private static string SanitizeConnectionString(string connectionString)
+        {
+            const string secretKey = "SharedAccessKey";
+            var parsed = ConnectionString.Parse(connectionString, allowEmptyValues: true);
+
+            // Configuration client expects secret to be base64 encoded so we can't use the placeholder
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/tests/TestConnectionStringSanitizer.cs:23]: this needs to be updated

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**